modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/account_selector/views/OrdersListView.js

```diff
@@ -90,10 +87,6 @@ function OrdersListView({
 
 			<ClayDropDown.Divider />
 
-			<li>
-				<div ref={ordersListRef} />
-			</li>
```

Comment:
This was a very strange way to do this. I need to properly test this in the UI but I really don't like this pattern being used. Effectively this was an anti-pattern here of bi-directional data flow. If the Autocomplete component was passed a `customView` prop, instead of rendering inside of the Autocomplete component it was handling all the business logic of requesting items and then rendering them via createPortal to this node. The reason this pattern is bad is because Autocomplete is now rendering to an unrelated area. If we want to properly do this, we must pull the state(items) up the react tree and outside of Autocomplete. That way we either pass the items to Autocomplete or we render them here.

Reply:
cc @FabioDiegoMastrorilli

I noticed you added this API in https://github.com/liferay/liferay-portal/commit/4aa38531329c324378a9181651407d55df3d9a5b#diff-fbc58da25a23f789ae15c90776dd7b5ce184557ea3d17369a143dfb8bb7554bcR270

Can you verify my changes in this PR and see how we can get around this pattern of using the portal to render the content on an outside node.

Reply:
Hi @bryceosterhaus it makes totally sense :) 
Passing a ref there would help in case we would like to use all the autocomplete functionalities (search, infinite scrolling, custom views) to render a list of results even out of a react context. In that case there won't be any state tree so it wont't look like an unsafe operation.

That being said... please have a look at the `AccountSelector` component. I've use the Autocomplete there to avoid code duplication and, if I could still use it there to get the same result, it would be great.